executor/logger/worker_test.go

```diff
@@ -0,0 +1,218 @@
+	}
+}
+
+func writeTmpFile(dirname string, filename string, contents []byte) *os.File {
+	f, err := ioutil.TempFile(dirname, filename)
+	if err != nil {
+		log.Fatalf("cannot create temporary file %s/%s: %v", dirname, filename, err)
+	}
+
+	_, err = f.Write(contents)
+	if err != nil {
+		log.Fatalf("cannot write to temporary file %s/%s: %v", dirname, filename, err)
+	}
+
+	return f
+}
+
+func TestWorkerProducerLog(t *testing.T) {
+	config := tlsConfig(t)
+	p, err := kafka.NewProducer(&kafka.ConfigMap{
+		"bootstrap.servers":        "broker",
+		"go.delivery.reports":      false,
+		"debug":                    "all",
+		"socket.timeout.ms":        10,
+		"security.protocol":        "ssl",
+		"ssl.ca.location":          config.CaPEM,
+		"ssl.key.location":         config.KeyPEM,
+		"ssl.certificate.location": config.CertPEM,
+		"ssl.key.password":         config.KeyPassword,
+	})
+
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	expectedLogs := map[struct {
+		tag     string
+		message string
+	}]bool{
+		{"INIT", "librdkafka"}: false,
+	}
+	go func() {
+		for {
+			select {
+			case log, ok := <-p.Logs():
+				if !ok {
+					return
+				}
+
+				t.Log(log.String())
+
+				for expectedLog, found := range expectedLogs {
+					if found {
+						continue
+					}
+					if log.Tag != expectedLog.tag {
+						continue
+					}
+					if strings.Contains(log.Message, expectedLog.message) {
+						expectedLogs[expectedLog] = true
+					}
+				}
+			}
+		}
+	}()
+}
+
+func TestWorkerProducerWrongFilesSSL(t *testing.T) {
+	config := tlsConfig(t)
+	_, err := kafka.NewProducer(&kafka.ConfigMap{
+		"security.protocol":        "SSL",
+		"ssl.ca.location":          "/fakepath/to/test-ca.pem",
+		"ssl.key.location":         config.KeyPEM,
+		"ssl.certificate.location": config.CertPEM,
+		"ssl.key.password":         config.KeyPassword,
+	})
+	_, err2 := kafka.NewProducer(&kafka.ConfigMap{
+		"security.protocol":        "SSL",
+		"ssl.ca.location":          config.CaPEM,
+		"ssl.key.location":         "/fakepath/to/test-access.key",
+		"ssl.certificate.location": config.CertPEM,
+		"ssl.key.password":         config.KeyPassword,
+	})
+	_, err3 := kafka.NewProducer(&kafka.ConfigMap{
+		"security.protocol":        "SSL",
+		"ssl.ca.location":          config.CaPEM,
+		"ssl.key.location":         config.KeyPEM,
+		"ssl.certificate.location": "/fakepath/to/test-cert.cert",
+		"ssl.key.password":         config.KeyPassword,
+	})
+	if err != nil && err2 != nil && err3 != nil {
+		t.Logf("Producers threw errors as expected %s", err)
+
+	}
+}
+
+func TestWorkerProducerCorrectFilesSSL(t *testing.T) {
+	config := tlsConfig(t)
+	_, err := kafka.NewProducer(&kafka.ConfigMap{
+		"security.protocol":        "SSL",
+		"ssl.ca.location":          config.CaPEM,
+		"ssl.key.location":         config.KeyPEM,
+		"ssl.certificate.location": config.CertPEM,
+		"ssl.key.password":         config.KeyPassword,
+	})
+
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+}
```

Comment:
:thought_balloon: It'd be a bit more idiomatic to use `t.Run` for nested tests/a test matrix, plus it makes the expectations a bit easier to interact with and understand:
```suggestion
func TestWorkerKafkaConfigurations(t *testing.T) {
	type test struct {
		name        string
		kafkaConfig kafka.ConfigMap
		expectError bool
	}

	g := NewWithT(t)
	config := tlsConfig(t)

	tests := []test{
		{
			"All options are valid",
			kafkaConfig: kafka.ConfigMap{
				"security.protocol":        "SSL",
				"ssl.ca.location":          config.CaPEM,
				"ssl.key.location":         config.KeyPEM,
				"ssl.certificate.location": config.CertPEM,
				"ssl.key.password":         config.KeyPassword,
			},
			expectError: false,
		},
		{
			name: "CA cert location is invalid",
			kafkaConfig: kafka.ConfigMap{
				"security.protocol":        "SSL",
				"ssl.ca.location":          "/fakepath/to/test-ca.pem",
				"ssl.key.location":         config.KeyPEM,
				"ssl.certificate.location": config.CertPEM,
				"ssl.key.password":         config.KeyPassword,
			},
			expectError: true,
		},
		{
			name: "Private key file location is invalid",
			kafkaConfig: kafka.ConfigMap{
				"security.protocol":        "SSL",
				"ssl.ca.location":          config.CaPEM,
				"ssl.key.location":         "/fakepath/to/test-access.key",
				"ssl.certificate.location": config.CertPEM,
				"ssl.key.password":         config.KeyPassword,
			},
			expectError: true,
		},
		{
			name: "Public key certificate location is invalid",
			kafkaConfig: kafka.ConfigMap{
				"security.protocol":        "SSL",
				"ssl.ca.location":          config.CaPEM,
				"ssl.key.location":         config.KeyPEM,
				"ssl.certificate.location": "/fakepath/to/test-cert.cert",
				"ssl.key.password":         config.KeyPassword,
			},
			expectError: true,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t testing.T) {
			_, err := kafka.NewProducer(&tt.kafkaConfig)
			if tt.expectError {
				g.Expect(err).ToNot(BeNil())
			} else {
				g.Expect(err).To(BeNil())
			}
		})
	}
}
```

Reply:
I imported `gomega` for your suggestion because `BeNil()` and `NewWithT()` were not defined and I think they are from this package.

https://pkg.go.dev/github.com/onsi/gomega#section-readme

Reply:
Ah, they can be imported as immediate identifiers by using `. "github.com/..."` (note the leading `. `) - hard to add multi-part suggestions in GitHub :upside_down_face: 
But yes - it's already used in the executor package.